src/Common/tests/System/Security/Cryptography/AlgorithmImplementations/ECDsa/ECDsaTestsBase.cs

```diff
@@ -0,0 +1,273 @@
+            }
+        }
+
+        private static IEnumerable<CurveDef> TestCurvesRaw
+        {
+            get
+            {
+                // nistP* curves
+                yield return new CurveDef()
+                {
+                    Curve = ECCurve.NamedCurves.nistP256, // also secp256r1
+                    KeySize = 256,
+                    CurveType = ECCurve.ECCurveType.PrimeShortWeierstrass,
+                    RequiredOnPlatform = true,
+                };
+                yield return new CurveDef()
+                {
+                    Curve = ECCurve.NamedCurves.nistP384, // also secp384r1
+                    KeySize = 384,
+                    CurveType = ECCurve.ECCurveType.PrimeMontgomery,
+                    RequiredOnPlatform = true,
+                };
+                yield return new CurveDef()
+                {
+                    Curve = ECCurve.NamedCurves.nistP521, // also secp521r1
+                    KeySize = 521,
+                    CurveType = ECCurve.ECCurveType.PrimeShortWeierstrass,
+                    RequiredOnPlatform = true,
+                };
+                yield return new CurveDef()
+                {
+                    Curve = ECCurve.NamedCurves.brainpoolP160r1,
+                    KeySize = 160,
+                    CurveType = ECCurve.ECCurveType.PrimeShortWeierstrass
+                };
+                yield return new CurveDef()
+                {
+                    Curve = ECCurve.CreateFromOid(new Oid("1.3.132.0.24", "")), // sect193r1
+                    KeySize = 193,
+                    CurveType = ECCurve.ECCurveType.Characteristic2,
+                };
+                yield return new CurveDef()
+                {
+                    Curve = ECCurve.CreateFromOid(new Oid("1.2.840.10045.3.0.1", "")), // c2pnb163v1
+                    KeySize = 163,
+                    CurveType = ECCurve.ECCurveType.Characteristic2,
+                };
+                yield return new CurveDef()
+                {
+                    Curve = ECCurve.CreateFromOid(new Oid("1.3.132.0.16", "")), // sect283k1
+                    KeySize = 283,
+                    CurveType = ECCurve.ECCurveType.Characteristic2,
+                };
+                yield return new CurveDef()
+                {
+                    Curve = ECCurve.CreateFromOid(new Oid("1.3.132.0.17", "")), // sect283r1
+                    KeySize = 283,
+                    CurveType = ECCurve.ECCurveType.Characteristic2,
+                };
+                yield return new CurveDef()
+                {
+                    Curve = ECCurve.CreateFromOid(new Oid("", "wap-wsg-idm-ecid-wtls7")),
+                    KeySize = 160,
+                    CurveType = ECCurve.ECCurveType.PrimeMontgomery,
+                };
+                yield return new CurveDef()
+                {
+                    Curve = ECCurve.CreateFromOid(new Oid("invalid", "invalid")),
```

Comment:
Okay, I think that something probed to see if this was supported, that failed (which set the last error), and then something else failed without setting an OpenSSL error.  So wherever we're probing to see if curves are supported we'll want to either call ERR_clear_error() (native) or build an OpenSslCryptographicException object (managed -- which will drain the error queue).  We're letting errors leak and confuse matters.